packages/block-editor/src/components/block-controls/use-has-block-controls.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+import { __experimentalUseSlotFills as useSlotFills } from '@wordpress/components';
+import warning from '@wordpress/warning';
+
+/**
+ * Internal dependencies
+ */
+import groups from './groups';
+
+export function useHasAnyBlockControls() {
+	let hasAnyBlockControls = false;
+	for ( const group in groups ) {
```

Comment:
Too clever? We can avoid this by duplicating the group names here. The con of doing that is that someone might add a new group and forget to update both files.